distribution/tools/entitlement-runtime/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyParserException.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.policy;
+
+import org.elasticsearch.xcontent.XContentLocation;
+
+public class PolicyParserException extends RuntimeException {
```

Comment:
I'm not sure what Elastic's philosophy is on runtime exceptions. This strikes me as a situation where this feature (the policy parser) will be used in very few places--probably just one--and in that place, the caller is going to need to catch this exception and do something about it anyway. That means the drawbacks of checked exceptions are insignificant, and making it a checked exception might help remind future devs of the sharp edges here.

Reply:
Most of our other parse exceptions are also runtime exceptions. I think I'd prefer to go with that precedent for now. If a policy fails, ES will not start anyway, so is should be pretty obvious to anyone adding code around this what happened.